lib/web_ui/lib/src/engine/html/render_vertices.dart

```diff
@@ -613,79 +744,100 @@ class _GlContext {
+      _glCanvas!.style
+        ..position = 'absolute'
+        ..width = '${cssWidth}px'
+        ..height = '${cssHeight}px';
     }
   }
 
-  static void dispose() {
+  void dispose() {
     _canvas = null;
+    _glCanvas = null;
+  }
+
+  /// Feature detects OffscreenCanvas.
+  static bool get supported =>
+      js_util.hasProperty(html.window, 'OffscreenCanvas');
+}
+
+/// Creates gl context from cached OffscreenCanvas for webgl rendering to image.
+class _GlContextCache {
+  static int _maxPixelWidth = 0;
+  static int _maxPixelHeight = 0;
+  static _GlContext? _cachedContext;
+  static _OffScreenCanvas? _offScreenCanvas;
+
+  static void dispose() {
     _maxPixelWidth = 0;
     _maxPixelHeight = 0;
-    _glCanvas = null;
     _cachedContext = null;
+    _offScreenCanvas?.dispose();
   }
 
-  html.OffscreenCanvas? get canvas => _canvas;
-
   static _GlContext? createGlContext(int widthInPixels, int heightInPixels) {
-    final bool isWebKit = (browserEngine == BrowserEngine.webkit);
-
-    if (_OffscreenCanvas.supported) {
-      final _OffscreenCanvas offScreenCanvas =
-          _OffscreenCanvas(widthInPixels, heightInPixels);
-      _cachedContext ??= _GlContext.fromOffscreenCanvas(offScreenCanvas.canvas!);
-      _cachedContext!.setViewportSize(widthInPixels, heightInPixels);
-      return _cachedContext;
+    if (widthInPixels > _maxPixelWidth || heightInPixels > _maxPixelHeight) {
+      _cachedContext?.dispose();
+      _cachedContext = null;
+      _offScreenCanvas = null;
+      _maxPixelWidth = math.max(_maxPixelWidth, widthInPixels);
+      _maxPixelHeight = math.max(_maxPixelHeight, widthInPixels);
+    }
+    _offScreenCanvas ??= _OffScreenCanvas(widthInPixels, heightInPixels);
+    if (_OffScreenCanvas.supported) {
+      _cachedContext ??=
+          _GlContext.fromOffscreenCanvas(_offScreenCanvas!._canvas!);
     } else {
-      // Allocate new canvas element is size is larger.
-      if (widthInPixels > _maxPixelWidth || heightInPixels > _maxPixelHeight) {
-        _glCanvas = html.CanvasElement(
-          width: widthInPixels,
-          height: heightInPixels,
-        );
-        _glCanvas!.className = 'gl-canvas';
-        final double cssWidth = widthInPixels / EngineWindow.browserDevicePixelRatio;
-        final double cssHeight = heightInPixels / EngineWindow.browserDevicePixelRatio;
-        _glCanvas!.style
-          ..position = 'absolute'
-          ..width = '${cssWidth}px'
-          ..height = '${cssHeight}px';
-        _maxPixelWidth = widthInPixels;
-        _maxPixelHeight = heightInPixels;
-        _cachedContext?.dispose();
-        _cachedContext = null;
-      }
-      _cachedContext ??= _GlContext.fromCanvas(_glCanvas!, isWebKit);
-      _cachedContext!.setViewportSize(widthInPixels, heightInPixels);
-      return _cachedContext;
+      _cachedContext ??= _GlContext.fromCanvas(_offScreenCanvas!._glCanvas!,
+          (browserEngine == BrowserEngine.webkit));
```

Comment:
Use `webGLVersion == 1` instead of `browserEngine == BrowserEngine.webkit`. Some browsers fallback to webgl1 for reasons other than webkit, e.g. older Android phones and Chrome on Windows 7.

Reply:
Done.